modules/combined/test/tests/beam_eigenstrain_transfer/master.i

```diff
@@ -0,0 +1,224 @@
+# Test for multi app vector postprocessor to aux variable transfer
+
+# Master App contains 2 beams, each fixed at one end and oriented
+# in the z direction. Each subApp contains a 2D model with the same
+# height as the length of the beam. A time varying temperature profile
+# is assigned to each 2D model and the resulting yy strain along the
+# height of the 2D model is transferred to the beam model using the
+# multi app transfer. The resulting displacement of the beam should be
+# same as that of the 2D models.
+
+[Mesh]
+  type = FileMesh
+  file = 2_beams.e
+[]
+
+[Variables]
+  [./disp_x]
+  [../]
+  [./disp_y]
+  [../]
+  [./disp_z]
+  [../]
+  [./rot_x]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+  [./rot_y]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+  [./rot_z]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+[]
+
+[BCs]
+  [./fixx1]
+    type = DirichletBC
+    variable = disp_x
+    boundary = 1
+    value = 0.0
+  [../]
+  [./fixy1]
+    type = DirichletBC
+    variable = disp_y
+    boundary = 1
+    value = 0.0
+  [../]
+  [./fixz1]
+    type = DirichletBC
+    variable = disp_z
+    boundary = 1
+    value = 0.0
+  [../]
+  [./fixr1]
+    type = DirichletBC
+    variable = rot_x
+    boundary = 1
+    value = 0.0
+  [../]
+  [./fixr2]
+    type = DirichletBC
+    variable = rot_y
+    boundary = 1
+    value = 0.0
+  [../]
+  [./fixr3]
+    type = DirichletBC
+    variable = rot_z
+    boundary = 1
+    value = 0.0
+  [../]
+[]
+
+[Kernels]
+  [./solid_disp_x]
```

Comment:
We should really add an Action like we have for the other StressDivergence kernels so you don't need to set up all of these (especially since we have 6 rather than 3 components).

It looks like we don't have that for the Truss ones either. Come to think of it, can we get the truss formulation by using a subset of the beam code?  It would be good to consolidate the truss and beam capabilities as much as possible (as long as it doesn't make the truss input more complex).